File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.java
import java.util.*;

class Solution {
    public String addSpaces(String s, int[] spaces) {
        StringBuilder result = new StringBuilder();
        Set<Integer> spaceSet = new HashSet<>();
        for (int space : spaces) {
            spaceSet.add(space);
        }

        for (int i = 0; i < s.length(); i++) {
            if (spaceSet.contains(i)) {
                result.append(" ");
            }
            result.append(s.charAt(i));
        }

        return result.toString();
    }
}

File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        int n = s.size(), m = spaces.size();
        string ans(n+m,' ');
        for(int i = 0,j=0 ; i < s.size();i++){
            if(j < spaces.size() && i == spaces[j])j++;
            ans[i+j] = s[i];
        }
        return ans;
    }
};